projects/RabbitMQ.Client/client/api/ConnectionFactory.cs

```diff
@@ -109,7 +109,7 @@ public sealed class ConnectionFactory : ConnectionFactoryBase, IConnectionFactor
         /// Corresponds to the <code>rabbit.max_message_size</code> setting.
         /// Note: the default is 0 which means "unlimited".
         /// </summary>
-        public const uint DefaultMaxMessageSize = 134217728;
+        public const uint DefaultMaxMessageSize = 536870912;
```

Comment:
The comments state the the default value is 0, i.e. unlimited. Should we enforce this value?

Reply:
Yes, we should use a limited default as of 6.6.0. It's high enough to likely not affect many.

Reply:
The default max message size for RabbitMQ `main` (i.e. 3.13) is the following...
```
C:\Users\lbakken\development\rabbitmq\rabbitmq-server [rabbitmq-server-9258 ≡]
> git grep max_message_size
deps/rabbit/BUILD.bazel:138:        {max_message_size, 134217728},
deps/rabbit/Makefile:120:           {max_message_size, 134217728},
deps/rabbit/priv/schema/rabbit.schema:927:{mapping, "max_message_size", "rabbit.max_message_size"
```

...same with `3.12.x`
```
C:\Users\lbakken\development\rabbitmq\rabbitmq-server_v3.12.x [v3.12.x ≡]
> git grep max_message_size
deps/rabbit/BUILD.bazel:135:        {max_message_size, 134217728},
deps/rabbit/Makefile:118:           {max_message_size, 134217728},
deps/rabbit/priv/schema/rabbit.schema:927:{mapping, "max_message_size", "rabbit.max_message_size",
```

@MarcialRosales @michaelklishin why did we decide to use 512MiB in this code?